basalt-monado: CUDA build failure

### Nixpkgs version

- Stable (25.05)

### Describe the bug

basalt-monado fails to build on the latest version of nixpkgs. Specifically while building basalt for monado for CUDA enabled systems.

### Steps to reproduce

```nix
services.monado.enable = true;
hardware.nvidia.package = pkgs.linuxPackages_6_14.nvidiaPackages.beta;
hardware.nvidia.nixpkgs.config.cudaSupport = true;
```
hardware required: NVidia GPU

### Expected behaviour

Compilation completes

### Screenshots

_No response_

### Relevant log output

```console
error: builder for '/nix/store/3wjxmc22j6sl7r93rad5bk0npskgnlln-basalt-monado-0-unstable-2024-06-21.drv' failed with exit code 1;
       last 30 log lines:
       > -- Detecting C compile features - done
       > -- Detecting CXX compiler ABI info
       > -- Detecting CXX compiler ABI info - done
       > -- Check for working CXX compiler: /nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/g++ - skipped
       > -- Detecting CXX compile features
       > -- Detecting CXX compile features - done
       > -- Dind't find ccache
       > -- Detected Linux with gcc.
       > -- Found Eigen3: /nix/store/rzbrwrvqvx10wpad0z9qf1hrxa0mmfyr-eigen-3.4.0-unstable-2022-05-19/include/eigen3 (Required is exact version "3.4.0")
       > -- Found Eigen headers in: /nix/store/rzbrwrvqvx10wpad0z9qf1hrxa0mmfyr-eigen-3.4.0-unstable-2022-05-19/include/eigen3
       > -- Found TBB 2021.11 (interface version 12110) headers in: /nix/store/vj7inkvjyd3s0r30h4b5pq81f4jlkffr-tbb-2021.11.0-dev/include
       > CMake Warning (dev) at /nix/store/vc5y8qhlnkq5ayr264l25n9490hlhrwf-opencv-4.11.0/lib/cmake/opencv4/OpenCVConfig.cmake:86 (find_package):
       >   Policy CMP0146 is not set: The FindCUDA module is removed.  Run "cmake
       >   --help-policy CMP0146" for policy details.  Use the cmake_policy command to
       >   set the policy and suppress this warning.
       >
       > Call Stack (most recent call first):
       >   /nix/store/vc5y8qhlnkq5ayr264l25n9490hlhrwf-opencv-4.11.0/lib/cmake/opencv4/OpenCVConfig.cmake:108 (find_host_package)
       >   CMakeLists.txt:294 (find_package)
       > This warning is for project developers.  Use -Wno-dev to suppress it.
       >
       > CMake Error at /nix/store/wphngc22a7aphbp5pi5jqmqlsqmisgn5-cmake-3.31.6/share/cmake-3.31/Modules/FindCUDA.cmake:883 (message):
       >   Specify CUDA_TOOLKIT_ROOT_DIR
       > Call Stack (most recent call first):
       >   /nix/store/vc5y8qhlnkq5ayr264l25n9490hlhrwf-opencv-4.11.0/lib/cmake/opencv4/OpenCVConfig.cmake:86 (find_package)
       >   /nix/store/vc5y8qhlnkq5ayr264l25n9490hlhrwf-opencv-4.11.0/lib/cmake/opencv4/OpenCVConfig.cmake:108 (find_host_package)
       >   CMakeLists.txt:294 (find_package)
       >
       >
       > -- Configuring incomplete, errors occurred!
       For full logs, run 'nix log /nix/store/3wjxmc22j6sl7r93rad5bk0npskgnlln-basalt-monado-0-unstable-2024-06-21.drv'.
error: 1 dependencies of derivation '/nix/store/sz28bhs1s5mzf4ps5bnrv3b3m1qaa99c-unit-monado.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xsh7l95ck1fkdakr6lf1vb5d1rmmisxa-user-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5fsn8ss1j8mgjzk8hfj5hg4hm14xwzb3-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mhcqpfbpqzgxhnn0kl74bs0c2ai02qxf-nixos-system-nixos-25.05.20250606.70c74b0.drv' failed to build
```

### Additional context


Ive attempted to solve the problem by declaring the cmake flag for the monado package, but it appears to not share the same cmake flags with basalt.
my attempt to solve it is as follows:
```nix
services.monado = {
  package = (pkgs.monado.overrideAttrs(oldAttrs:{
    cmakeFlags = oldAttrs.cmakeFlags ++ [
      (lib.cmakeFeature "CUDA_TOOLKIT_ROOT_DIR" "${pkgs.cudaPackages.cudatoolkit}")
    ];
  }));
};
```
do note, this didnt change a thing.

Edit by @Scrumplex: this issue was initially reported for Monado while this is actually an issue with basalt-monado. Disregard the above.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.10, NixOS, 25.05 (Warbler), 25.05.20250606.70c74b0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: aarch64-linux, i686-linux, riscv64-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/lotus/.config/nix/nix.conf:/nix/store/qlyxsdy64q30fqkzwrxfdsijcwl2spj3-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/lotus/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/lotus/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/lotus/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/wb91j9c6gm6fqn30jxc4jf2nfs4mb4s8-lix-2.91.1/share`
 - nixpkgs: `/nix/store/x1lsj3bnjq6n909mjsnr6maf632bg41m-source`

### Notify maintainers

@Scrumplex @prusnak

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ba/basalt-monado/package.nix
+++ b/pkgs/by-name/ba/basalt-monado/package.nix
@@ -1,4 +1,5 @@
 {
+  config,
   autoPatchelfHook,
   boost,
   bzip2,
@@ -21,6 +22,8 @@
   stdenv,
   tbb,
   xorg,
+  cudaPackages,
+  enableCuda ? config.cudaSupport,
 }:
 stdenv.mkDerivation {
   pname = "basalt-monado";
@@ -63,6 +66,7 @@ stdenv.mkDerivation {
     (lib.cmakeBool "BASALT_INSTANTIATIONS_DOUBLE" false)
     (lib.cmakeBool "BUILD_TESTS" false)
     (lib.cmakeFeature "EIGEN_ROOT" "${eigen}/include/eigen3")
+    (lib.optionals enableCuda "-DCUDA_TOOLKIT_ROOT_DIR=${cudaPackages.cudatoolkit}")
   ];
 
   passthru.updateScript = nix-update-script { };

